resources/assets/js/videos/components/videoScreen/drawInteractions.vue

```diff
@@ -163,6 +164,23 @@ export default {
             this.$emit('pending-annotation', null);
         },
         extendPendingAnnotation(e) {
+            // Check polygons
+            if (e.feature.getGeometry().getType() === 'Polygon') {
+                let polygonValidator = new PolygonValidator(e.feature);
+                if (polygonValidator.isInvalidPolygon()) {
+                    // Disallow polygons with less than three non-overlapping points
+                    this.$emit('is-invalid-polygon')
+                    this.removeFeature(e.feature)
+                    return;
+                }
+
+                // If polygon is self-intersecting, create valid polygon
+                let validPolygon = polygonValidator.getValidPolygon();
```

Comment:
The method names are a little unintuitive IMO. Why does it fail if the polygon is "invalid" if there is a method that can make it "valid" later? Maybe call the second method "cleanPolygon" or something like that.